test/Tester/DependencyInjectionDisambiguationTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Orleans;
+using Orleans.Runtime.Configuration;
+using Orleans.TestingHost;
+using TestExtensions;
+using UnitTests.GrainInterfaces;
+
+namespace UnitTests.General
+{
+    [TestCategory("DI")]
+    public class DependencyInjectionDisambiguationTests : OrleansTestingBase, IClassFixture<DependencyInjectionDisambiguationTests.Fixture>
+    {
+        private readonly Fixture fixture;
+
+        public class Fixture : BaseTestClusterFixture
+        {
+            protected override TestCluster CreateTestCluster()
+            {
+                var options = new TestClusterOptions();
+                options.ClusterConfiguration.UseStartupType<MyStartup>();
+                return new TestCluster(options);
```

Comment:
Thought it would be good to test this working in a grain, but fair point.  Test can be much simpler.  Will simplify